### PR TITLE
MNT Add GitHub Actions CI

### PR DESCRIPTION
**Pull request by emteknetnz**
DO NOT SQUASH MERGE

Issue https://github.com/silverstripe/gha-ci/issues/37

Includes backport cherry-pick of https://github.com/silverstripe/silverstripe-framework/pull/10334/commits/ec5b94facf9ea438c47524c4f07d2ecdf36a901c to fix https://github.com/silverstripe/silverstripe-framework/runs/7441583580?check_suite_focus=true `3) SilverStripe\ORM\Tests\MySQLPDOConnectorTest::testConnectionCharsetControl with data set #1 ('utf8', 'utf8_general_ci', 'utf8_unicode_520_ci')`

On `Utf8TestHelper::dbVersion` had to change private `?string` property to a mixed var for php 7.3 support, though it's a private on a TestOnly class so it's fine to merge-up

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    - cron: '20 14 * * 2'
+
+jobs:
+  ci:
+    uses: silverstripe/gha-ci/.github/workflows/ci.yml@v1
+    with:
+      # Turn phpcoverage off because it causes a segfault
+      phpcoverage_force_off: true
+      # There is a strange behat.yml file in framework that runs behat tests in the admin module
+      endtoend: false
```
Comment (GuySartorelli) [inline .github/workflows/ci.yml:18]: If we're not running those tests, shouldn't we just remove the behat file instead?

File: tests/php/ORM/DataObjectSchemaGenerationTest.php [hunk 2]
```diff
@@ -67,11 +68,26 @@ function () {
         );
     }
 
+    private function isMySQL8(): bool
+    {
+        $connector = DB::get_conn()->getConnector();
+        if ($connector instanceof MySQLiConnector &&
+            preg_match('#^8\.#', $connector->getVersion())
```
Comment (GuySartorelli) [inline tests/php/ORM/DataObjectSchemaGenerationTest.php:75]: This won't work for mariadb, but this is just a temporary measure as per the TODO comments below, so this is fine.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (emteknetnz) [reply .github/workflows/ci.yml:18]: Removed

**Review by GuySartorelli — APPROVED** [review]
Only change from [the last run](https://github.com/silverstripe/silverstripe-framework/actions/runs/2714930099) is changing how we're removing behat tests. We can see the behat test isn't running, and the last run was fully green so we know this one will be green as well. Merging.

**Pull request merged**